doc/reference/data_structures/dlist.rst

```diff
@@ -12,38 +12,38 @@ pointers per node, and thus has somewhat higher runtime code and
 memory space needs.
 
 A ``sys_dlist_t`` struct may be instantiated by the user in any
-accessible memory.  It must be initialized with ``sys_dlist_init()``
-or ``SYS_DLIST_STATIC_INIT()`` before use.  The ``sys_dnode_t`` struct
+accessible memory.  It must be initialized with :c:func:`sys_dlist_init`
+or :c:macro:`SYS_DLIST_STATIC_INIT` before use.  The ``sys_dnode_t`` struct
```

Comment:
do `c:type:` or `c:struct:` work for stuff like `sys_dnote_t` (and others)?

Reply:
they could, if they are available through doxygen though.

By the way, I noticed that we do not get a failure when we use :c:func:`...` on something that is not defined in doxygen, is that normal?

Reply:
I don't know, that could be the way Sphinx C domain handles missing references (maybe https://github.com/sphinx-doc/sphinx/blob/4.x/sphinx/domains/c.py#L3840 ?).